src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/MarlinInterface.java

```diff
@@ -110,23 +106,30 @@ private void setupNetworkListener() {
 	
 	// This does not fire on the Swing EVT thread.  Be careful!  Concurrency problems may happen.
 	protected void onDataReceived(NetworkSessionEvent evt) {
-		if(evt.flag == NetworkSessionEvent.DATA_RECEIVED) {
-			lastReceivedTime=System.currentTimeMillis();
+		if (evt.flag == NetworkSessionEvent.DATA_RECEIVED) {
+			lastReceivedTime = System.currentTimeMillis();
 			String message = ((String)evt.data).trim();
-			//logger.debug("MarlinInterface received '"+message.trim()+"'.");
-			if(message.startsWith(STR_OK)) {
+
+			if (logger.isTraceEnabled() && !message.startsWith(STR_OK)) {
+				logger.trace("received '{}'", message.trim());
+			}
+
+			if (message.startsWith(STR_OK)) {
 				onHearOK();
-			} else if(message.contains(STR_RESEND)) {
+			} else if (message.contains(STR_RESEND)) {
+				logger.trace("received '{}'", message.trim());
```

Comment:
this will duplicate the previous logger.trace

Reply:
good catch